src/sidebar-counters/sidebar-counters.js

```diff
@@ -4,66 +4,80 @@
+export const menus = [HREF_BRANCHES, HREF_PULL_REQUESTS]
 
-    a.style.position = 'relative'
+const MAX_COUNTER = 99
 
-    const size =
-        // eslint-disable-next-line eqeqeq, no-eq-null
-        resources == null || resources.size == null
-            ? '?'
-            : resources.size > 999999
-                ? '999999+'
-                : resources.size
+export let menusCounters = {
+    HREF_BRANCHES: null,
+    HREF_PULL_REQUESTS: null,
+}
+
+type ResultSize = $Exact<{ size: number }>
 
-    const badge = (
+export function getBadge(size: number): HTMLElement {
+    const maxReached = size > MAX_COUNTER
+    return (
         <span class="__rbb-badge">
-            <span class="__rbb-badge-counter">{size}</span>
+            <span class={`__rbb-badge-counter${maxReached ? ' -max' : ''}`}>
+                {typeof size !== 'number' ? '!' : Math.min(size, MAX_COUNTER)}
+            </span>
         </span>
     )
-    a.insertBefore(badge, a.firstChild)
-    return badge
 }
 
-export default async function addSideBarCounters() {
-    // Exit early if can't find any of the nav links
-    // that we are going to augment with badge counters
-    const branchesLink: HTMLAnchorElement = await elementReady(
-        'a[href$="branches/"]'
-    )
-    const pullRequestsLink: HTMLAnchorElement = await elementReady(
-        'a[href$="pull-requests/"]'
-    )
-
-    if (!branchesLink || !pullRequestsLink) {
-        return
+async function getCounterInfo(wantedMenuFromHref: string): Promise {
+    switch (wantedMenuFromHref) {
+        case HREF_BRANCHES:
+            return api.getBranches()
+        case HREF_PULL_REQUESTS:
+            return api.getPullrequests()
+        default:
+            return
     }
+}
 
-    const [branches, pullrequests] = await Promise.all([
-        api.getBranches(),
-        api.getPullrequests(),
-    ])
+export async function addCounterToMenuItem(menu: HTMLElement) {
+    const link: HTMLAnchorElement = menu.querySelector('a')
 
-    const branchesBadge = addBadge(branchesLink, branches)
-    const prBadge = addBadge(pullRequestsLink, pullrequests)
+    if (!link) return
 
-    const resizeButton: HTMLButtonElement = await elementReady(
-        '.ak-navigation-resize-button'
+    const hrefParts = link.getAttribute('href').split('/')
+    const wantedMenuFromHref = hrefParts.find(x => menus.includes(x))
+
+    // Exit early if can't find one of the nav links
+    // that we are going to augment with badge counters
+    if (!wantedMenuFromHref) return
+
+    const size = menusCounters[wantedMenuFromHref]
+    const badge = getBadge(size)
+
+    menu.style.overflow = 'hidden'
+    link.style.position = 'relative'
+    link.insertBefore(badge, link.firstChild)
+}
+
+export default async function addSideBarCounters() {
+    menus.forEach(async menu => {
+        const resources = await getCounterInfo(menu)
+        const { size: currentSize } = resources || {}
+        menusCounters[menu] = currentSize
+    })
```

Comment:
If the network calls take longer to resolve than it takes the page to mount the sidebar component, the user will not see the counts in the badge.

I think we should await these calls before proceeding with the rest of the execution so that we guarantee that the data is available before trying to add the badge. This is what the original implementation was doing i this part:

```tsx
    const [branches, pullrequests] = await Promise.all([
        api.getBranches(),
        api.getPullrequests(),
    ])
```
This would also remove the need to have the `getCounterInfo` function, simplifying things a little bit.

Do you think we can fallback to the same behavior, or did you have a specific reason to change this that I'm not seeing?